fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -6,6 +6,7 @@
 import pandas as pd
 import pickle
 import scipy.optimize as opt
+from time import time
```

Comment:
Rather than changing these methods, would it be better to wrap the estimators themselves? I can see arguments either way.

Reply:
Not entirely sure what you mean here

Reply:
Basically, do we think that the profiling facility should be added to `fairlearn` itself, or should the performance framework wrap the supplied estimators and do the profiling that way. You'd have something like:
```python
class PerformanceEstimator:
   def __init__(estimator):
      self.real_estimator = estimator
      self.n_fit_calls = 0
      self.n_fit_call_elapsed = TimeSpan()


    def fit(…..):
       self.n_fit_calls++
      Stopwatch.start()
      self.real_estimator.fit()
      Stopwatch.stop()
      self.n_fit_call_elapsed += Stopwatch.elapsed
```
Like I said, there are arguments either way; I'm just noting the alternative.

Reply:
Got it!
I think this may be something users are interested in tracking as well, and it's not really a lot of overhead to keep a few floats around.
What's the benefit of Stopwatch over the time package?

Reply:
`Stopwatch` was pseudo-code :-)

As for what accumulates the profiling information.... arguments both ways as I said. If we are going to put it into the shipped `fairlearn` code, can I suggest creating a `ProfileInformation` (or similar class), and returning that as a property on the Result object. That way, we don't have to worry about steadily adding more fields in multiple places.

Reply:
The complication there is that even expgrad and grid search aren't consistent since expgrad just has a series of calls to fit while grid search has separate result objects. The takeaway from our in-person discussion was that we can always adjust this later if we feel strongly either way. This is not exposed in any way that would make users think they can rely on it existing since we used underscores.